config/volume.go

```diff
@@ -343,35 +347,34 @@ func (c *Client) WatchSnapshotSignal(activity chan *watch.Watch) {
 	watch.Create(w)
 }
 
-// Validate options for a volume. Should be called anytime options are
-// considered.
-func (ro *RuntimeOptions) Validate() error {
-	if ro.UseSnapshots && (ro.Snapshot.Frequency == "" || ro.Snapshot.Keep == 0) {
-		return errored.Errorf("Snapshots are configured but cannot be used due to blank settings")
-	}
-
-	return nil
-}
-
-// Validate validates a volume configuration, returning error on any issue.
+// Validate validates a volume configuration, returning error on any
 func (cfg *Volume) Validate() error {
-	if cfg.Backends.Mount == "" {
-		return errored.Errorf("Mount volume cannot be blank for volume %v", cfg)
+	if err := ValidateVolume(cfg); err != nil {
+		return errors.ErrJSONValidation.Combine(err)
 	}
 
-	if cfg.VolumeName == "" {
-		return errored.Errorf("Volume Name was omitted for volume %v", cfg)
+	do, err := cfg.ToDriverOptions(time.Second)
+	if err != nil {
+		return err
 	}
 
-	if cfg.PolicyName == "" {
-		return errored.Errorf("Policy name was omitted for volume %v", cfg)
-	}
+	// We use a few dummy variables to ensure that global configuration is
+	// not needed in the storage drivers, that the validation does not fail
+	// because of it.
+	var mountPath string
+	for driverType, backendName := range map[string]string{backend.Mount: cfg.Backends.Mount, backend.CRUD: cfg.Backends.CRUD, backend.Snapshot: cfg.Backends.Snapshot} {
+		defer func() { mountPath = "" }()
```

Comment:
don't set variables in defer; also I don't think this will fire on time for it to be useful. You're also doing it in a loop which will allocate a function for each entry, which I don't believe is necessary.